SalesforceCommerceCloud/pwa-kit
# Add wishlist item to the cart

# Description



This PR allows adding wishlist items to the cart.

This PR was suppose to be a small change but turns out needing to refactor a few components include `AddToCartModal` and `ProductView`. The reason is the `AddToCartModal` was embedded in `ProductView` as a local component. This works okay until we need to have `ProductView` in a modal. In the below screen, you will see a situation where we need to display the `AddToCartModal` after successfully adding item to the cart from the `ProductView` modal.

<img width="1229" alt="Screen Shot 2021-10-05 at 3 53 48 PM" src="https://user-images.githubusercontent.com/10948652/136451073-925a95c4-244a-4761-b050-b334cc299491.png">

Because the component hierarchy looks like this:
```
<ProductViewModal>
  <ProductView>
    <AddToCartModal />
  </ProductView>
</ProductViewModal>
```

This is limiting the ability to display the two modals separately. We need to support the situation where `AddToCartModal` is rendered on the page while `ProductViewModal` is not. 

The implementation becomes really messy after i'm trying to add a new `AddToCartModal` in the wishlist components.

So, I decided to move `AddToCartModal` to top level as a global component, as this modal is shown on multiple pages like pdp, cart and wishlist. All place share the same modal. To achieve that, i also need to create a context to store the modal open/close state.

# Types of Changes



- [] **Bug fix** (non-breaking change that fixes an issue)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Move addtocartmodal out of ProductView

# How to Test-Drive This PR

- `npm start`
- Go to http://localhost:3000
- Add a few items to your wishlist, include both master product and variant products
- Then go to http://localhost:3000/en-GB/account/wishlist , click add to cart on your variant item
- verify the item has been successfully added to cart
- click "select option" on a master item, and use the modal to add item to the cart, verify you can do so successfully.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
